### ISSUE TITLE
Rolling back in API adds a new record in the history collection

### ISSUE THREAD

**Issue by ericln** [emirotin/mongodb-migrations#13]
I am seeing this behaviour when the `rollback` is called after the `migrate`, it still adds the error script into the history collection. That basically will cause the next run to skip the failed script and continue with the rest of the scripts.

following is the example I have

Migration script

``` js
module.exports.id = "12";

module.exports.up = function (done) {
  var coll = this.db.collection('test');
  coll.insert({test:"test 12"}, function(err, data){
    return done('12 has error');
  });
};

module.exports.down = function (done) {
  var coll = this.db.collection('test');
  coll.remove({}, done);
};
```

Main program that runs the migration

``` js
async.series([
  function run(callback){
    migrator.migrate(
      function(err, data){
        if(err){
          console.log('migration error', err);
          return callback('error');
        }
        console.log('migration completed');
        callback();
      }
    );
  }
], function(err) {
  if (err) {
    console.log(err);
    migrator.rollback(function (err) {
      if (err) console.log('Rollback Error', err);
      console.log('Rollback complete');
      process.exit();
    });
  }
  else {
    process.exit();
  }
});
```

**Comment by emirotin** [comment]
Ah, so it seems that it's rollback that makes the records which indeed doesn't make sense.
I'll try to find some time to fix it, but I have to be honest: I didn't pay attention to this library since I stopped working with MongoDB some time ago.

**Comment by ericln** [comment]
arh, i see at least it's not something to do with how we use the api. it'll be great if this can be fix when you have a moment. not really familiar with coffeescript myself

**Comment by eberhara** [comment]
@emirotin Can you add to the docs that you DO NOT support this project anymore?
Also mention that it does not support .rollback() properly.

Thanks!

**Comment by emirotin** [comment]
It's been more than a year but I have finally properly investigated and fixed this issue.
It will be released soon as part of the 0.8.0 release.
Note that this changes the behavior (even though erroneous) so should be considered the breaking change.
